pagination/paginator.go

```diff
@@ -83,10 +86,10 @@ func Paginator[T any](ctx context.Context, listFunc ListFunc[T], processFunc Pro
 		}
 
 		if !shouldContinue {
-			break
+			return allItems, ErrPartialResultsPossible
 		}
```

Comment:
Is `shouldContinue` just a boolean? On `ErrPartialResultsPossible`, wouldn't it make more sense to introduce a custom error type that the user can unwrap in order to get more data?